### PR TITLE
Zen keyboard update for Rev2

### PR DESCRIPTION
**Pull request by XScorpion2**
Refactor of the Zen keyboard directory to support Rev2 for @Legonut 

## Description
Moving data around into appropriate folders.
Switched to split common code over custom matrix
Minor changes to support #5477 & #5288 when / if the land (if they don't land, will update this keyboard at that time)
All features tested on Zen rev1 & Zen r2 and confirmed working. (including optional pr #5477 & #5288 features)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Legonut** [comment]
I've been working with @XScorpion2 on this, tested and works on Zen Rev1 and Rev2 hardware.

File: keyboards/zen/rev1/config.h [hunk 1]
```diff
@@ -54,8 +50,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 /* Locking resynchronize hack */
 #define LOCKING_RESYNC_ENABLE
 
-#define BACKLIGHT_PIN F4
-#define BACKLIGHT_LEVELS 5
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (drashna) [inline keyboards/zen/rev1/config.h]: Could you remove this?  

This is the default, and shouldn't be needed.

File: keyboards/zen/rev2/config.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (drashna) [inline keyboards/zen/rev2/config.h]: Could this be removed? 

It's the default value, and shouldn't be needed.

File: keyboards/zen/rev2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
+#include QMK_KEYBOARD_H
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/zen/rev2/keymaps/default/keymap.c]: This shouldn't be needed. it should be included by the keyboard's header file, which is defined as `QMK_KEYBOARD_H` at compile time. 

```suggestion
```

File: keyboards/zen/rev2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    _______, RGB_SAI, RGB_VAI, RGB_SAD, RESET,   KC_LBRC,                   KC_RBRC, KC_PGUP,  KC_UP,   KC_PGDN, KC_INS,  KC_HOME, \
+    _______, RGB_HUD, RGB_VAD, RGB_HUI, _______, _______,                   _______, KC_LEFT,  KC_DOWN, KC_RGHT, KC_DEL,  KC_END,  \
+    KC_LSFT, _______, _______, _______, _______, _______, _______, _______, _______, _______,  _______, KC_MPLY, KC_MPRV, KC_MNXT, \
+    KC_LCTL, KC_LGUI, KC_LALT, RGB_MOD, _______, _______, _______, _______, _______, _______,  _______, KC_MUTE, KC_VOLU, KC_VOLD, \
+                                                 _______, _______, _______, _______
+  ),
+
+};
+
+#ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/zen/rev2/keymaps/default/keymap.c]: Could this block be removed?  It's not needed,  as `set_single_persistent_default_layer` will actually handle this.

File: keyboards/zen/rules.mk [hunk 5]
```diff
@@ -1,51 +1,26 @@
-SRC += matrix.c \
-	   split_util.c \
-	   serial.c
+ifneq ($(strip $(USE_PROTON_C)),)
```
Comment (drashna) [inline keyboards/zen/rules.mk]: https://docs.qmk.fm/#/proton_c_conversion

File: keyboards/zen/rules.mk [hunk 6]
```diff
@@ -57,17 +32,13 @@ EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
...
-BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
 MIDI_ENABLE = no            # MIDI controls
 AUDIO_ENABLE = no           # Audio output on port C6
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
-SUBPROJECT_rev1 = yes
-USE_I2C = no
-# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SUBPROJECT_rev2 = yes
```
Comment (drashna) [inline keyboards/zen/rules.mk]: This doesn't do anything

```suggestion
```

File: keyboards/zen/rev2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,14 @@
+ENCODER_ENABLE = yes
+
+OLED_DRIVER_ENABLE ?= no
```
Comment (drashna) [inline keyboards/zen/rev2/rules.mk]:
```suggestion
OLED_DRIVER_ENABLE = no
```

File: keyboards/zen/rev2/rules.mk [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+OLED_ROTATE90 ?= yes
```
Comment (drashna) [inline keyboards/zen/rev2/rules.mk]:
```suggestion
OLED_ROTATE90 = yes
```

File: keyboards/zen/rev1/config.h (hunk 1 above)
Comment (XScorpion2) [reply keyboards/zen/rev1/config.h]: Removed

File: keyboards/zen/rev2/config.h (hunk 2 above)
Comment (XScorpion2) [reply keyboards/zen/rev2/config.h]: Removed

File: keyboards/zen/rev2/keymaps/default/keymap.c (hunk 3 above)
Comment (XScorpion2) [reply keyboards/zen/rev2/keymaps/default/keymap.c]: Removed

File: keyboards/zen/rev2/keymaps/default/keymap.c (hunk 4 above)
Comment (XScorpion2) [reply keyboards/zen/rev2/keymaps/default/keymap.c]: Removed

File: keyboards/zen/rules.mk (hunk 5 above)
Comment (XScorpion2) [reply keyboards/zen/rules.mk]: Fixed, `CONVERT_TO_PROTON_C=yes` or `CTPC=yes` now compile successfully. Untested as Proton-C orders are backlogged.

File: keyboards/zen/rules.mk (hunk 6 above)
Comment (XScorpion2) [reply keyboards/zen/rules.mk]: Removed

File: keyboards/zen/rev2/rules.mk (hunk 8 above)
Comment (XScorpion2) [reply keyboards/zen/rev2/rules.mk]: Removed

File: keyboards/zen/rev2/rules.mk (hunk 7 above)
Comment (XScorpion2) [reply keyboards/zen/rev2/rules.mk]: Removed

File: keyboards/zen/rev2/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      tap_code(KC_DOWN);
+    }
+  }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/zen/rev2/keymaps/default/keymap.c]: You don't need the bit shifting here.  The function handles that, actually. 

```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

Comment (XScorpion2) [reply keyboards/zen/rev2/keymaps/default/keymap.c]: fixed

**Comment by XScorpion2** [comment]
Force pushed rebase on latest master so folks with Zen rev2 can get their split encoder wigglies on!

**Comment by XScorpion2** [comment]
Any remaining issues on this PR @drashna ?

**Comment by drashna** [comment]
Nope, looks good!

**Pull request merged**